src/pgroonga-primary-maintainer.c

```diff
@@ -209,6 +213,83 @@ pgroonga_primary_maintainer_wal_size_check_all(void)
 	pgstat_report_activity(STATE_IDLE, NULL);
 }
 
+static bool
+checkPGroongaPrimaryMaintainerHours(char **newval,
+									void **extra,
+									GucSource source)
+{
+	char *rawstring;
+	List *elemlist;
+	ListCell *l;
+	int i = 0;
+	bool hours[HOURS_SIZE] = {};
+
+	for (i = 0; i < HOURS_SIZE; i++)
+	{
+		hours[i] = false;
+	}
+
+	if (*newval == NULL || *newval[0] == '\0')
+		return true;
+
+	rawstring = pstrdup(*newval);
+	if (!SplitDirectoriesString(rawstring, ',', &elemlist))
+	{
+		list_free_deep(elemlist);
+		pfree(rawstring);
+		ereport(LOG,
+				(errcode(ERRCODE_SYNTAX_ERROR),
+				 errmsg("invalid syntax in pgroonga_primary_maintainer.hour")));
+		return false;
+	}
+
+	foreach (l, elemlist)
+	{
+		char *hourString = (char *) lfirst(l);
+		int hour = atoi(hourString);
+		if ((strncmp(hourString, "0", 1) != 0 && hour == 0) ||
+			(hour < 0 || hour > 23))
+		{
+			list_free_deep(elemlist);
+			pfree(rawstring);
+			ereport(
+				LOG,
+				(errcode(ERRCODE_DATATYPE_MISMATCH),
+				 errmsg("invalid syntax in pgroonga_primary_maintainer.hour")));
+			return false;
+		}
+
+		hours[hour] = true;
+	}
+
+	list_free_deep(elemlist);
+	pfree(rawstring);
+
+	// for 16 or higher
+	// *extra = guc_malloc(LOG, HOURS_SIZE * sizeof(bool));
+	*extra = malloc(HOURS_SIZE * sizeof(bool));
+	memcpy(*extra, hours, HOURS_SIZE * sizeof(bool));
```

Comment:
We don't want to allocate this.
Can we define `PGroongaPrimaryMaintainerHours` as `static bool PGroongaPrimaryMaintainerHours[HOURS_SIZE]` instead?
We can re-parse the user input in the assign function.